legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperMappingBuilder.java

```diff
@@ -508,4 +519,91 @@ public static void buildMappingClassOutOfLocalProperties(SetImplementation setIm
             }
         }
     }
+
+
+    public static Test processMappingTestSuite(org.finos.legend.engine.protocol.pure.v1.model.test.Test test, CompileContext context)
+    {
+        if (test instanceof MappingTestSuite)
+        {
+            MappingTestSuite mappingTestSuite = (MappingTestSuite) test;
+            Root_meta_pure_mapping_metamodel_MappingTestSuite pureMappingTestSuite = new Root_meta_pure_mapping_metamodel_MappingTestSuite_Impl("");
+
+            if (mappingTestSuite.tests == null || mappingTestSuite.tests.isEmpty())
+            {
+                throw new EngineException("Mapping TestSuites should have atleast 1 test", mappingTestSuite.sourceInformation, EngineErrorType.COMPILATION);
+            }
+
+            List<String> testIds = ListIterate.collect(mappingTestSuite.tests, t -> t.id);
+            List<String> duplicateTestIds = testIds.stream().filter(e -> Collections.frequency(testIds, e) > 1).distinct().collect(Collectors.toList());
+
+            if (!duplicateTestIds.isEmpty())
+            {
+                throw new EngineException("Multiple tests found with ids : '" + String.join(",", duplicateTestIds) + "'", mappingTestSuite.sourceInformation, EngineErrorType.COMPILATION);
+            }
+
+            RichIterable<? extends Root_meta_pure_test_AtomicTest> tests = ListIterate.collect(mappingTestSuite.tests, unitTest -> (Root_meta_pure_test_AtomicTest) HelperMappingBuilder.processMappingTestSuite(unitTest, context));
+
+            pureMappingTestSuite._id(test.id);
+            if (mappingTestSuite.storeTestDatas != null)
+            {
+                pureMappingTestSuite._storeTestDatas(ListIterate.collect(mappingTestSuite.storeTestDatas, ele -> HelperMappingBuilder.processMappingElementTestData(ele, context, new ProcessingContext("Mapping Element: "))));
+            }
+            else
+            {
+                throw new EngineException("Test data needs to be provided for a valid test");
+            }
+            pureMappingTestSuite._tests(tests);
+
+            return pureMappingTestSuite;
+        }
+        else if (test instanceof MappingTest)
+        {
+            MappingTest mappingTest = (MappingTest) test;
+            Root_meta_pure_test_AtomicTest pureMappingTest = context.getCompilerExtensions().getExtraTestProcessors().stream().map(processor -> (Root_meta_pure_test_AtomicTest) processor.value(mappingTest, context, new ProcessingContext("Mapping Test '" + mappingTest.id + "' Second Pass")))
```

Comment:
why we need to go via extensions ?

Reply:
mapping tests lie as part of the core compiler extensions

Reply:
Resolved